### PR TITLE
[MAINTENANCE] type-checking `checkpoint/actions.py`

### PR DESCRIPTION
**Pull request by Kilo59**
Type check `checkpoint/actions.py`
There are no runtime changes in this PR.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/checkpoint/actions.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def __init__(
         self.notify_with = notify_with
         self.show_failed_expectations = show_failed_expectations
 
-    def _run(
+    def _run(  # type: ignore[override] # signature does not match parent
```
Comment (cdkini) [inline great_expectations/checkpoint/actions.py:240]: 😢 how can we remedy this issue? It feels like a violation of sound OOO design and I'm quite confident there are probably some latent bugs around as a result of it.

File: great_expectations/checkpoint/actions.py [hunk 2]
```diff
@@ -1182,7 +1181,7 @@ def _run(
         # <snippet name="great_expectations/checkpoint/actions.py get_docs_sites_urls">
         docs_site_urls_list = self.data_context.get_docs_sites_urls(
             resource_identifier=validation_result_suite_identifier,
-            site_names=self._site_names,
+            site_names=self._site_names,  # type: ignore[arg-type] # ._site_names could be None
```
Comment (cdkini) [inline great_expectations/checkpoint/actions.py:1184]: Could we instead do `self._site_names or []`? (assuming this is a nullable list or collection)

**Review by cdkini — APPROVED** [review]

File: great_expectations/checkpoint/actions.py (hunk 2 above)
Comment (Kilo59) [reply great_expectations/checkpoint/actions.py:1184]: Good call, that sounds right. I'll test that out, and do it as part of this final `checkpoint` typing PR.
* https://github.com/great-expectations/great_expectations/pull/7901

File: great_expectations/checkpoint/actions.py (hunk 1 above)
Comment (Kilo59) [reply great_expectations/checkpoint/actions.py:240]:
```python
   # parent
    def _run(
        self,
        validation_result_suite: ExpectationSuiteValidationResult,
        validation_result_suite_identifier: Union[
            ValidationResultIdentifier, GXCloudIdentifier
        ],
        data_asset,
        expectation_suite_identifier=None,
        checkpoint_identifier=None,
    ):
```
```python
    # child
    def _run(  # type: ignore[override] # signature does not match parent
        self,
        validation_result_suite: ExpectationSuiteValidationResult,
        validation_result_suite_identifier: Union[
            ValidationResultIdentifier, GXCloudIdentifier
        ],
        data_asset=None,
        payload=None,
        expectation_suite_identifier=None,
        checkpoint_identifier=None,
    ):
```

Comment (Kilo59) [reply great_expectations/checkpoint/actions.py:240]: The difference is `data_asset` not being required and the addition of the optional `payload`.
So we could either add `payload` to the parent signature, and make `data_asset` optional.... or we could replace all the individual parameters with a single `options`-like type. 

We already have an issue with a lot of these methods only use a few of these parameters. This particular method only uses `payload` (pulse the required params).

https://refactoring.guru/smells/long-parameter-list

**Pull request merged**